folding-schemes/src/folding/nova/decider_eth.rs

```diff
@@ -355,28 +370,63 @@ pub mod tests {
             N,              // here we define the FoldingScheme to use
         >;
 
-        let mut rng = ark_std::test_rng();
+        let mut rng = rand::rngs::OsRng;
         let poseidon_config = poseidon_canonical_config::<Fr>();
 
         let F_circuit = CubicFCircuit::<Fr>::new(()).unwrap();
         let z_0 = vec![Fr::from(3_u32)];
 
-        let prep_param = PreprocessorParam::new(poseidon_config, F_circuit);
-        let nova_params = N::preprocess(&mut rng, &prep_param).unwrap();
+        let preprocessor_param = PreprocessorParam::new(poseidon_config, F_circuit);
+        let nova_params = N::preprocess(&mut rng, &preprocessor_param).unwrap();
 
         let start = Instant::now();
-        let mut nova = N::init(&nova_params, F_circuit, z_0.clone()).unwrap();
+        let nova = N::init(&nova_params, F_circuit, z_0.clone()).unwrap();
         println!("Nova initialized, {:?}", start.elapsed());
+
+        // prepare the Decider prover & verifier params
+        let (decider_pp, decider_vp) = D::preprocess(&mut rng, &nova_params, nova.clone()).unwrap();
+
+        // serialize the Nova params. These params are the trusted setup of the commitment schemes used
+        // (ie. KZG & Pedersen in this case)
+        let mut nova_pp_serialized = vec![];
+        nova_params
+            .0
+            .serialize_compressed(&mut nova_pp_serialized)
+            .unwrap();
+        let mut nova_vp_serialized = vec![];
+        nova_params
+            .1
+            .serialize_compressed(&mut nova_vp_serialized)
+            .unwrap();
+        // deserialize the Nova params. This would be done by the client reading from a file
+        let nova_pp_deserialized = NovaProverParams::<
+            Projective,
+            Projective2,
+            KZG<'static, Bn254>,
+            Pedersen<Projective2>,
+        >::deserialize_compressed(
+            &mut nova_pp_serialized.as_slice()
+        )
+        .unwrap();
+        let nova_vp_deserialized = NovaVerifierParams::<
+            Projective,
+            Projective2,
+            KZG<'static, Bn254>,
+            Pedersen<Projective2>,
+        >::deserialize_compressed(
+            &mut nova_vp_serialized.as_slice()
+        )
+        .unwrap();
+
+        // initialize nova again, but from the deserialized parameters
+        let nova_params = (nova_pp_deserialized, nova_vp_deserialized);
+        let mut nova = N::init(&nova_params, F_circuit, z_0).unwrap();
```

Comment:
Can't we just have a sepparated test done for serialization rather than including it into another test?
I think is not only more clear and clean. But makes easier to identify if we get tests failing if the serialization was broken or the error is another one.

Reply:
I was afraid of repeating almost the same test twice where one of them is just the same as the other but with extra serialization-deserialization logic, but agree that better to almost-duplicate it while keeping simplicity in this case. Updated.